operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,331 @@
+        origin_ll = reversed(dset_coords[0][0])  # Grab origin directly from database
+        origin = projectLcc(*origin_ll)
+
+        lat, lon = self._lat_lon
+        coords = (lon, lat)
+        coords = projectLcc(*coords)
+        delta = np.subtract(coords, origin)
+        ij = [int(round(x/2000)) for x in delta]
+        return tuple(reversed(ij))
+
+    def ws_diurnal_prep(self, start_date = '2007-01-01', end_date = '2013-12-31'):
+
+        """
+        This method returns a Pandas Series corresponding to hourly average windspeeds
+
+        Args:
+        start_date(:obj:'String'): start date to diurnal analysis (optional)
+        end_date(:obj:'String'): end date to diurnal analysis (optional)
+
+
+        Returns:
+        ws_diurnal (Pandas Series): Series where each index corresponds to a different hour of the day and each value corresponds to the average windspeed
+        """
+
+        f = h5pyd.File("/nrel/wtk-us.h5", 'r')
+
+        # Setup date and time
+        dt = f["datetime"]
+        dt = pd.DataFrame({"datetime": dt[:]},index=range(0,dt.shape[0]))
+        dt['datetime'] = dt['datetime'].apply(dateutil.parser.parse)
+
+        project_idx = self.indicesForCoord(f)
+
+        print("y,x indices for project: \t\t {}".format(project_idx))
+        print("Coordinates of project: \t {}".format(self._lat_lon))
+        print("Coordinates of project: \t {}".format(f["coordinates"][project_idx[0]][project_idx[1]]))
+
+        # Get wind speed at 80m from the specified lat/lon
+        ws = f['windspeed_80m']
+        t_range = dt.loc[(dt.datetime >= start_date) & (dt.datetime < end_date)].index
+    
+        # Convert to dataframe
+        ws_tseries = ws[min(t_range):max(t_range)+1, project_idx[0], project_idx[1]]
+        ws_df=pd.DataFrame(index=dt.loc[t_range,'datetime'],data={'ws':ws_tseries})
+    
+        # Calculate diurnal profile of wind speed
+        ws_diurnal=ws_df.groupby(ws_df.index.hour).mean()
+
+        return ws_diurnal
+
+    def wtk_diurnal_plot(self):
+
+        """
+        This method plots the WTK diurnal plot alongisde the hourly power averages of the scada_df across all turbines
+        Args:
+        (None)
+        Returns: (None)
+        """
+
+        sum_df = self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+        df_temp = sum_df.copy()
+        df_temp[self._t] = df_temp.index
+
+        df_diurnal = df_temp.groupby(df_temp[self._t].dt.hour)[self._w].mean()
+
+        ws_norm = self._wtk_ws_diurnal/self._wtk_ws_diurnal.mean()
+        df_norm = df_diurnal/df_diurnal.mean()
+    
+        plt.figure(figsize=(8,5))
+        plt.plot(ws_norm, label = 'WTK wind speed (UTC)')
+        plt.plot(df_norm, label = 'QC power')
+        plt.grid()
+        plt.xlabel('Hour of day')
+        plt.ylabel('Normalized values')
+        plt.title('WTK and QC Timezone Comparison')
+        plt.legend()
+        plt.show()
+
+    def corr_df_calc(self):
```

Comment:
This seems to print out the correlation series in the reverse order (i.e. highest correlations in the 19, 20 range, and not the 4 or 5 range. I think it's more intuitive to have the latter (i.e., looks like a 5 hour shift to get the data into UTC).